src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updatelistingrequirements/UpdateListingRequirementsHandler.java

```diff
@@ -0,0 +1,210 @@
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        return response;
+    }
+
+    @NotNull
+    public List<DynamicListItem> generateInterpreterLanguage() {
+
+        return getLanguages().stream()
+            .map(this::getLanguageDynamicListItem)
+            .collect(Collectors.toList());
+    }
+
+    @NotNull
+    public DynamicListItem getLanguageDynamicListItem(Language language) {
+        String reference = language.getReference();
+        String name = language.getNameEn();
+
+        if (nonNull(language.getDialectReference())) {
+            reference = String.format("%s-%s", language.getReference(), language.getDialectReference());
+            name = language.getDialectEn();
+        }
+        return new DynamicListItem(reference, name);
+    }
+
+    public List<Language> getLanguages() {
+        List<Language> signLanguages = signLanguagesService.getSignLanguages();
+        List<Language> verbalLanguages = verbalLanguagesService.getVerbalLanguages();
+
+        return Stream.concat(signLanguages.stream(), verbalLanguages.stream())
+            .collect(Collectors.toList());
+    }
+
+    public List<DynamicListItem> generateReservedMembers() {
+        JudicialRefDataUsersRequest request = JudicialRefDataUsersRequest.builder()
+            .ccdServiceName(SERVICE_NAME)
+            .build();
+
+        JudicialRefDataUsersResponse refDataResponse = judicialRefData.getJudicialUsers(
+            idamService.getIdamTokens().getIdamOauth2Token(),
+            idamService.getIdamTokens().getServiceAuthorization(),
+            request);
+
+        return Optional.ofNullable(refDataResponse.getJudicialUsers())
+            .orElse(Collections.emptyList())
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(judicialUser -> isNotEmpty(judicialUser.getAppointments()))
+            .filter(judicialUser -> judicialUser.getAppointments().stream().map(JudicialMemberAppointments::getAppointment).anyMatch(this::isValidJudicialMemberType))
+            .map(this::getJudicialMemberListItem)
+            .collect(Collectors.toList());
+    }
+
+    @NotNull
+    public DynamicListItem getJudicialMemberListItem(JudicialUser judicialUser) {
+        String referenceCodes = String.format("%s|%s", judicialUser.getPersonalCode(), extractHmcReferenceCode(judicialUser));
+
+        String name = isNotBlank(judicialUser.getPostNominals())
+            ? String.format("%s %s", judicialUser.getFullName(), judicialUser.getPostNominals())
+            : judicialUser.getFullName();
+        return new DynamicListItem(referenceCodes, name);
+    }
+
+    private String extractHmcReferenceCode(JudicialUser judicialUser) {
+        if (judicialUser.getAppointments() != null) {

```

Comment:
I maybe wrong here but could this be flipped to look abit cleaner such as bring everything outside of the if (judicialUser.getAppointments() != null) If not looks good to me.